src/cmd/flux-mini.py

```diff
@@ -0,0 +1,305 @@
+        parser.add_argument(
+            "-N", "--nodes", type=int, help="Number of nodes to allocate"
+        )
+        parser.add_argument(
+            "-n", "--ntasks", type=int, default=1, help="Number of tasks to start"
+        )
+        parser.add_argument(
+            "-c",
+            "--cpus-per-task",
+            type=int,
+            default=1,
+            help="Number of cpus to allocate per task",
+        )
+        parser.add_argument(
+            "-t", "--time", help="Time limit in Flux standard duration: e.g. 2d or 1.5h"
+        )
+        parser.add_argument(
+            "--priority",
+            help="Set job priority (0-31, default=16)",
+            type=int,
+            default=16,
+        )
+        parser.add_argument("command", nargs=argparse.REMAINDER)
+        return parser
+
+    def submit(self, args):
+        """
+        Submit job, constructing jobspec from args.
+        Returns jobid.
+        """
+        jobspec = JobSpec(
+            args.command,
+            num_tasks=args.ntasks,
+            cores_per_task=args.cpus_per_task,
+            num_nodes=args.nodes,
+        )
+        jobspec.set_cwd(os.getcwd())
+        jobspec.set_environment(dict(os.environ))
+        if args.time is not None:
+            jobspec.set_duration(args.time)
+
+        h = flux.Flux()
+        return job.submit(h, jobspec.dumps(), priority=args.priority)
+
+    def main(self, args):
+        jobid = self.submit(args)
+        print(jobid, file=sys.stdout)
+
+    def get_parser(self):
+        return self.parser
+
+
+class RunCmd(SubmitCmd):
+    """
+    RunCmd is identical to SubmitCmd, except it attaches the the job
+    after submission.  Some additional options are added to modify the
+    attach behavior.
+
+    Usage: flux mini run [OPTIONS] cmd ...
+    """
+
+    def __init__(self):
+        self.parser = self.create_parser()
+        self.parser.add_argument(
+            "--label",
+            action="store_true",
+            help="Add rank labels to stdout, stderr lines",
+        )
+        self.parser.add_argument(
+            "-v", "--verbose", action="count", help="Increase verbosity on stderr"
+        )
+
+    def main(self, args):
+        jobid = self.submit(args)
+        if args.verbose > 0:
+            print("jobid:", jobid, file=sys.stderr)
+
+        # Attach to job to display stdout, stderr
+        attach_args = ["flux-job", "attach"]
+        if args.label:
+            attach_args.append("--label")
+        if args.verbose > 1:
+            attach_args.append("--show-events")
+        if args.verbose > 2:
+            attach_args.append("--show-exec")
+
+        attach_args.append(str(jobid))
+        os.execvp("flux-job", attach_args)
+
+
+logging.basicConfig(level=logging.INFO, format="flux-mini: %(levelname)s %(message)s")
+logger = logging.getLogger(__name__)
+
+
+@util.CLIMain(logger)
+def main():
+    parser = argparse.ArgumentParser()
```

Comment:
maybe set `prog="flux mini"` for argparse so that help doesn't reference `flux-mini.py` which breaks the 4th wall of the `flux(1)` command driver ;-)

Reply:
Ok, applying the fix to the util class discussed in #2234, and setting the name to "flux-mini".

Reply:
Was this pushed, still seeing:
```console
$ flux mini 
usage: flux-mini.py [-h] {run,submit} ...
flux-mini.py: error: too few arguments
```

This is minor enough that it could wait for a future pr.

Reply:
Sorry, my my comment was about the logging prefix.  I had not set the `ArgumentParser(prog="flux-mini")`.  I can also override the usage string to print something more like our standard boiler plate for commands with subcommands here.

Fixup coming.

Reply:
Ah, sorry, my misinterpretation. I see now that it is obvious what you meant.